.cbmc-batch/jobs/aws_cryptosdk_sig_get_privkey/aws_cryptosdk_sig_get_privkey_harness.c

```diff
@@ -23,28 +23,26 @@
 #include <cipher_openssl.h>
 
 void aws_cryptosdk_sig_get_privkey_harness() {
-    /* arguments */
-    struct aws_cryptosdk_sig_ctx *ctx = can_fail_malloc(sizeof(struct aws_cryptosdk_sig_ctx));
+    /* Nondet Input */
+    struct aws_cryptosdk_sig_ctx *ctx = ensure_sig_ctx_has_allocated_members();
     struct aws_allocator *alloc       = can_fail_allocator();
     struct aws_string *priv_key;
 
-    /* assumptions */
-    __CPROVER_assume(ctx);
-    ensure_sig_ctx_has_allocated_members(ctx);
+    /* Assumptions */
     __CPROVER_assume(aws_cryptosdk_sig_ctx_is_valid_cbmc(ctx));
     __CPROVER_assume(ctx->is_sign);  // context has to be in signing mode, otherwise private key is NULL
 
-    /* operation under verification */
+    /* Operation under verification */
     if (aws_cryptosdk_sig_get_privkey(ctx, alloc, &priv_key) == AWS_OP_SUCCESS) {
```

Comment:
Can `priv_key` be null?

Reply:
Yes, it can!

Reply:
But `&priv_key` will always point to something. Is that what we want?

Reply:
We're taking `priv_key` as a `**`, so we need to check if either level of the indirection is null

Reply:
I've allocated priv_key like so  
 `  struct aws_string *priv_key       = ensure_string_is_allocated_nondet_length();`
and then assumed that it's either NULL or a valid string and added that precondition to the source code. priv_key is immediately set to NULL in the function 
`*priv_key = NULL;`. 
However, when I allocate this way or through malloc() as above, I get a memory leak error even though I call free(priv_key) at the end of the function. I'm not very used to using the  --memory-leak-check flag, so any idea why this might be?

Reply:
I think the memory-leak check is not worth it, personally.

Reply:
I've removed it.